### ISSUE TITLE
Visual Studio crashes

### ISSUE THREAD

**Issue by kuechlerm** [laurentkempe/GitDiffMargin#96]
As of today (version 3.1.1.0) this extention causes Visual Studio 2015 to crash if I open a project that uses git.
Debugging VS shows me this error message:

```
System.TypeInitializationException was unhandled
Message: An unhandled exception of type 'System.TypeInitializationException' occurred in LibGit2Sharp.dll
Additional information: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception.
```

Disabling the extention stops the crashing, but I much rather would enable it ;)

**Comment by laurentkempe** [comment]
@kuechlerm Thanks for the report! Could you try to [install this version](https://github.com/laurentkempe/GitDiffMargin/releases/tag/v3.1.2) and let me know if you still have issues!

We have several other users reporting crashes due to LibGit2Sharp and Libgit2  #93 #92 #87

**Comment by kuechlerm** [comment]
That version is working :+1:

**Comment by drewnoakes** [comment]
I also just experienced a crash. Actually Windows Defender flagged `git2-785d8c4.dll` as containing `Trojan:Win32/Varpes.N!cl` at the same time that VS crashed.

**Comment by martinskuta** [comment]
Yeah as a workaround you can exclude devnev.exe from Windows devender, until libgit2 and windows defender get it right.

**Comment by laurentkempe** [comment]
@drewnoakes Two of my colleagues are just reporting to me that they had the same issue with the update of Windows Defender signature of today! I have Visual Studio folder excluded from Windows Defender so I didn't get it.

**Comment by PavelAgejkin** [comment]
thanks, v3.1.2 solved crashing issue.
please update visualstudiogallery ASAP because it took me 4 hours to find cause of crashing VS.
I'm sure I'm not alone with this issue.

Thanks for nice extension and have a nice day

**Comment by shiftkey** [comment]
@drewnoakes @laurentkempe what is the version of the Antispyware definitions you're running that are triggering this?

**Comment by laurentkempe** [comment]
@glconti Could you send a screenshot of the version here please!

I have the following but just to be sure there are the same

![image](https://cloud.githubusercontent.com/assets/272612/14885852/1804951a-0d4e-11e6-8bda-a17435c66da3.png)

**Comment by glconti** [comment]
@laurentkempe here there is
![image](https://cloud.githubusercontent.com/assets/13535297/14885901/4e90b686-0d4e-11e6-977a-73c539cee6f8.png)

**Comment by laurentkempe** [comment]
@PavelAgejkin I don't want to update at the moment because it uses a not released version of LibGit2Sharp!

**Comment by PavelAgejkin** [comment]
@shiftkey same version as above 1.219.115.0

@laurentkempe OK.
Can you make extension just not to load, but not to crash VS?

**Comment by laurentkempe** [comment]
> Can you make extension just not to load, but not to crash VS?

@PavelAgejkin I am not sure I get that!

**Comment by PavelAgejkin** [comment]
@laurentkempe 
Now if Your extension crashes, it crashes VS ( as **DllNotFoundException** bubbles to VS).
I offer to catch that **DllNotFoundException** and to handle it (e.g. disable extension or to show alert or to print exception message to Output window)

**Comment by laurentkempe** [comment]
@PavelAgejkin I see. So and so at the moment, I think the best is to either remove devenv.exe from Windows defender scan or to uninstall the extension.

But for the next released version I will think about something like your proposal!

**Comment by drewnoakes** [comment]
@laurentkempe I have the same defender definitions as you.

+1 for fixing the extension on the gallery such that it doesn't crash VS at boot. That's a very bad experience, no matter the reason. Any fix will do. Even just display a dialog that explains the potential for the problem and a workaround before crashing.

**Comment by laurentkempe** [comment]
@PavelAgejkin @drewnoakes So the crash you have is with 3.1.1 and is linked with Windows Defender?
If you can confirm that 3.1.2 avoid the problem then I will update it on the gallery!

**Comment by drewnoakes** [comment]
3.1.2 fixed the issue for me, thanks.

**Comment by PavelAgejkin** [comment]
@laurentkempe yes, Windows Defender removes DLL and VS cant start, becase DLL is not found.
3.1.2 resolved issue
Event viewer shows this error:
> Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.DllNotFoundException
   at LibGit2Sharp.Core.NativeMethods.git_libgit2_init()
   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor()
   at LibGit2Sharp.Core.NativeMethods..cctor()

> Exception Info: System.TypeInitializationException
   at LibGit2Sharp.Core.NativeMethods.RemoveHandle()
   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject.Finalize()

**Comment by laurentkempe** [comment]
@drewnoakes @PavelA85 That's it I have uploaded a fix version on Visual Studio Gallery!

**Comment by laurentkempe** [comment]
Fixed with 3.1.2
